src/app/shared/spinner/spinner.component.ts

```diff
@@ -0,0 +1,39 @@
+import {Component, Input, OnDestroy} from 'angular2/core';
+
+@Component({
+  selector: 'spinner',
+  styleUrls: ['app/shared/spinner/spinner.component.css'],
+  template: `<div [hidden]="!isSpinning" class="spinner"></div>`
+})
+
+export class SpinnerComponent implements OnDestroy {
+
+  private currentTimeout: number;
+  private isSpinning: boolean = false;
+
+  @Input() public delay: number = 300;
+
+  @Input() public set spinning(value: boolean) {
+    if (!value) {
+      this.cancelTimeout();
+      this.isSpinning = false;
+    }
+    if (this.currentTimeout) {
+      return;
+    }
+    this.currentTimeout = setTimeout(() => {
```

Comment:
I have only read about these, so this is just a question not a critique, but is there any advantage to using an rxjs future scheduler?

https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/schedulers/scheduler.md#rxschedulerprototypeschedulefuturestate-duetime-action